src/Generators/Google.php

```diff
@@ -14,26 +14,16 @@ class Google implements Generator
     /** @var string {@see https://www.php.net/manual/en/function.date.php} */
     protected $dateFormat = 'Ymd';
     /** @var string */
-    protected $dateTimeFormat = 'Ymd\THis\Z';
+    protected $dateTimeFormat = 'Ymd\THis';
 
     /** {@inheritDoc} */
     public function generate(Link $link): string
     {
         $url = 'https://calendar.google.com/calendar/render?action=TEMPLATE';
 
-        $utcStartDateTime = (clone $link->from)->setTimezone(new DateTimeZone('UTC'));
-        $utcEndDateTime = (clone $link->to)->setTimezone(new DateTimeZone('UTC'));
         $dateTimeFormat = $link->allDay ? $this->dateFormat : $this->dateTimeFormat;
-        $url .= '&dates='.$utcStartDateTime->format($dateTimeFormat).'/'.$utcEndDateTime->format($dateTimeFormat);
-
-        // Add timezone name if it is specified in both from and to dates and is the same for both
-        if (
-            $link->from->getTimezone() && $link->to->getTimezone()
-            && $link->from->getTimezone()->getName() === $link->to->getTimezone()->getName()
-        ) {
-            $url .= '&ctz=' . $link->from->getTimezone()->getName();
-        }
-
+        $url .= '&dates='.$link->from->format($dateTimeFormat).'/'.$link->to->format($dateTimeFormat);
+        $url .= '&ctz=' . $link->from->getTimezone()->getName();
```

Comment:
This keeps the original TZ and passes it to google.
Also, it seems the `\Z` in $dateTimeFormat causes issues with full dates with time and Google thinks dates are given in UTC.